### PR TITLE
Add composite fields to aggregation

### PR DESCRIPTION
**Pull request by Weiko**
## Context
This PR introduces a first aggregation for a composite field

## Test
Screenshot 2024-11-15 at 15 37 05

### CODE REVIEW

File: packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts [hunk 1]
```diff
@@ -19,17 +15,18 @@ export class ProcessAggregateHelper {
     for (const [aggregatedFieldName, aggregatedField] of Object.entries(
       selectedAggregatedFields,
     )) {
-      const fieldMetadata = fieldMetadataMapByName[aggregatedField.fromField];
-
-      if (!fieldMetadata) {
+      if (
+        !aggregatedField?.fromColumnName ||
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts:19]: I'm not a big fan of these tests not being done on boolean.
I would prefer gate such as isUndefined

File: packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts [hunk 2]
```diff
@@ -19,17 +15,18 @@ export class ProcessAggregateHelper {
...
+        !aggregatedField?.aggregationOperation
+      ) {
         continue;
       }
 
-      const fieldName = fieldMetadata.name;
+      const columnName = aggregatedField.fromColumnName;
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts:25]: i would not talk about columns at this level (api). I think we should talk about subFields

File: packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts (hunk 1 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts:19]: Sure

File: packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts (hunk 2 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts:25]: @charlesBochet we are building a SQL request here so I think it makes sense to talk about columns?

Comment (charlesBochet) [reply packages/twenty-server/src/engine/api/graphql/graphql-query-runner/helpers/process-aggregate.helper.ts:25]: I don't agree! The ORM is building SQL, but here we are above, we are building things to put in the ORM (~entity schemas that are close to metadata), no?

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**